bluemira/equilibria/find.py

```diff
@@ -725,6 +728,124 @@ def get_flux_loop(psi_norm):
     return lcfs, separatrix
 
 
+def get_legs(equilibrium, n_layers: int, dx_off: float):
+    """
+    Get the legs of a separatrix.
+
+    Parameters
+    ----------
+    equilibrium: Equilibrium
+        Equilibrium for which to find the separatrix legs
+    n_layers: int
+        Number of flux surfaces to extract for each leg
+    dx_off: float
+        Total span in radial space of the flux surfaces to extract
+
+    Returns
+    -------
+    legs: Dict[str, List[Loop]]
+        Dictionary of the legs with each key containing a list of geometries
+
+    Raises
+    ------
+    EquilibriaError: if a strange number of legs would be found for an X-point
+
+    Notes
+    -----
+    Will return two legs in the case of a single null
+    Will return four legs in the case of a double null
+
+    We can't rely on the X-point being contained within the two legs, due to
+    interpolation and local minimum finding tolerances.
+    """
+
+    def extract_leg(flux_line, x_cut, z_cut):
+        radial_line = Loop(x=[x_cut - delta, x_cut + delta], z=[z_cut, z_cut])
+        arg_inters = join_intersect(flux_line, radial_line, get_arg=True)
+        arg_inters.sort()
+        # Lower null vs upper null
+        func = operator.lt if z_cut < o_point.z else operator.gt
+
+        if len(arg_inters) > 2:
+            EquilibriaError(
+                "Unexpected number of intersections with the separatrix around the X-point."
+            )
+
+        flux_legs = []
+        for arg in arg_inters:
+            if func(flux_line.z[arg + 1], flux_line.z[arg]):
+                leg = Loop.from_array(flux_line[arg:])
+            else:
+                leg = Loop.from_array(flux_line[: arg + 1])
+
+            # Make the leg flow away from the plasma core
+            if leg.argmin((x_cut, z_cut)) > 3:
+                leg.reverse()
+
+            flux_legs.append(leg)
+        if len(flux_legs) == 1:
+            flux_legs = flux_legs[0]
+        return flux_legs
+
+    def extract_offsets(ref_leg, direction=-1):
+
+        offset_legs = []
+        for dx in dx_offsets:
+            x, z = ref_leg.x[0] + direction * dx, ref_leg.z[0]
+            xl, zl = find_flux_surface_through_point(
+                equilibrium.x,
+                equilibrium.z,
+                equilibrium.psi(),
+                x,
+                z,
+                equilibrium.psi(x, z),
+            )
+            offset_legs.append(extract_leg(Loop(x=xl, z=zl), x, z))
+        return offset_legs
+
+    o_points, x_points = equilibrium.get_OX_points()
+    o_point = o_points[0]
+    x_points = x_points[:2]
+    separatrix = equilibrium.get_separatrix()
+    delta = equilibrium.grid.dx
+    n_layers = max(2, n_layers)
```

Comment:
Should we be honouring the number of layers that were passed into the function? I'd probably be a bit surprised if I'd asked for a single layer but got two - especially as this is not documented in the doc string.

Reply:
yeah fair does, actually probably one is OK too, but will require a bunch of checks. I will try and get this less smelly too.